### PR TITLE
Use babel for typescript compilation with minimal changes

### PR DESCRIPTION
**Pull request by chuckdries**
Kicking off #1737 🚀

If the goal is to exactly match existing behavior, why fix what isn't broken? This change uses babel's typescript preset to compile typescript without any dramatic changes to the existing rollup-babel flow, and invokes `tsc` directly to check types only.

I included a little test.ts with a type error in it to demonstrate that module resolution and type checking both work. I tried turning on checkJs in the tsconfig but typescript does not like that one bit. We've got our work cut out for us :)

I do not know why there are so many deletions in the package.lock file

~Edit: I just turned off my computer for the night but I realized I forgot to make it actually emit the type definitions. Sorry about the premature PR.~

### CODE REVIEW

**Comment by markerikson** [comment]
So just to confirm, the build failure is "intentional" for now, right? :)

**Comment by chuckdries** [comment]
Yes, we should remove test.ts before we merge. I got it emitting types and typescript code in all 3 build modes, but eslint is still unhappy

File: .eslintrc [hunk 1]
```diff
@@ -1,14 +1,20 @@
 {
-  "parser": "babel-eslint",
+  "parser": "@typescript-eslint/parser",
   "extends": [
     "eslint:recommended",
     "plugin:import/recommended",
     "plugin:react/recommended",
     "plugin:prettier/recommended"
+    // "plugin:@typescript-eslint/recommended"
```
[the remark this answers is not in the document]
Comment (markerikson) [reply .eslintrc:8]: Oh yeah, my experience has been that ruleset is way to over-opinionated.

We don't want to be focusing on lint rules right now. The immediate goal is just to get stuff _compiling_ at all, and then get most of the existing types into place. Linting would be a last step.

**Comment by markerikson** [comment]
Seems like a reasonable starting point. Let's get this in!

**Pull request merged**